lib/input_validation.rb

```diff
@@ -0,0 +1,64 @@
+class NilError < StandardError
+end
+class EmptyError < StandardError
+end
+
+def check_nil?(input)
```

Comment:
It would have been nice if you'd put these helpers into a `module` since having these floating out in the global namespace could cause some trouble.

```ruby
module InputValidation
  def self.check_nil?(input)
    # ...
  end
  # ...
end
```

You'd then just use it like:
```ruby
InputValidation.check_nil?(value)
```